src/units/experiments.grids.pas

```diff
@@ -229,32 +228,88 @@ function IntToCell(AN: Integer) : TCell;
 
 { TGrid }
 
-procedure TGrid.SetGridType(AGridType: TGridType);
+procedure TGrid.SetGridStyle(AGridStyle: TGridStyle);
 begin
-  if FGridStyle = AGridType then Exit;
-  FGridStyle := AGridType;
-  case AGridType of
+  if FGridStyle = AGridStyle then Exit;
+  FGridStyle := AGridStyle;
+  case AGridStyle of
     gtCircle : FGrid := GetCircularCentralGrid(FSeed, FCellsSize);
     gtSquare : FGrid := GetCentralGrid(FSeed, FCellsSize, DispersionStyle);
     gtDistributed: FGrid := GetCentralGrid(FSeed, FCellsSize, DispersionStyle);
   end;
 end;
 
-procedure TGrid.RandomizeGridList;
+procedure TGrid.RandomizeGridList(AGridList: TGridList);
 var
   i : integer;
 begin
-  InvalidateGridList;
-  for i := FCellsCount - 1 downto 0 do
-    FGridList.Exchange(i, RandomRange(0, i + 1));
+  for i := AGridList.Count - 1 downto 0 do
+    AGridList.Exchange(i, RandomRange(0, i + 1));
+  {$IFDEF DEBUG}
+  //for i in FGridList do WriteLn(i);
+  {$ENDIF}
 end;
 
-procedure TGrid.InvalidateGridList;
+function TGrid.InvalidateGridList(IsSample: Boolean): TGridList;
 var
   i : integer;
 begin
-  FGridList.Clear;
-  for i := 0 to FCellsCount - 1 do FGridList.Add(i);
+  Result:= TGridList.Create;
+  {
+    3x3
+    0..1..2
+    3..4..5
+    6..7..8
+  }
+  case FGridOrientation of
+      goNone: begin
+        for i := 0 to FCellsCount - 1 do Result.Add(i);
+      end;
+      goLeftToRight: begin
+        if IsSample then begin
+          Result.Add(0);
+          Result.Add(3);
+          Result.Add(6);
+        end else begin
+          Result.Add(2);
+          Result.Add(5);
+          Result.Add(8);
+        end;
+      end;
+      goRightToLeft: begin
+        if IsSample then begin
+          Result.Add(2);
+          Result.Add(5);
+          Result.Add(8);
+        end else begin
+          Result.Add(0);
+          Result.Add(3);
+          Result.Add(6);
+        end;
+      end;
+      goBottomToTop: begin
+        if IsSample then begin
+          Result.Add(6);
+          Result.Add(7);
+          Result.Add(8);
+        end else begin
+          Result.Add(0);
+          Result.Add(1);
+          Result.Add(2);
+        end;
+      end;
+      goTopToBottom: begin
+        if IsSample then begin
+          Result.Add(0);
+          Result.Add(1);
+          Result.Add(2);
+        end else begin
+          Result.Add(6);
+          Result.Add(7);
+          Result.Add(8);
+        end;
+      end;
+    end;
```

Comment:
A implementação atual é por força bruta para matrizes 3 x 3. Infelizmente ainda não sabemos como abstrair essa funcionalidade considerando:

- matrizes com tamanhos diferentes
- matrizes com estilos de grid diferentes (circular, etc).